internal/impl/devbox.go

```diff
@@ -764,6 +764,10 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	// Append variables from current env if --pure is not passed
 	currentEnv := os.Environ()
 	env, err := d.parseEnvAndExcludeSpecialCases(currentEnv)
+	env["DEVBOX_PROJECT_ROOT"] = d.projectDir
+	env["DEVBOX_CONFIG_DIR"] = d.projectDir + "/devbox.d"
+	env["DEVBOX_PACKAGES_DIR"] = d.projectDir + "/.devbox/virtenv/.wrappers"
+
```

Comment:
A couple of concerns:
1. We should insert this _after_ the `err != nil` check below.
2. We make a copy of the env on line 786 into `originalEnv`. So we likely want to add it some place after that.

My suggestion is add it before line 864 `configEnv, err := d.configEnvs(ctx, env)` so it can be used to derive other env-vars etc. in the devbox.json config

Reply:
Good suggestion, updated!